ingestion/src/metadata/ingestion/source/pipeline/airflow/metadata.py

```diff
@@ -408,11 +409,12 @@ def yield_pipeline_lineage_details(
             )
         )
 
-        for task in dag.tasks:
-            for from_fqn in self.get_inlets(task) or []:
+        xlets = get_xlets_from_dag(dag=dag)
+        for xlet in xlets:
+            for from_fqn in xlet.inlets:
                 from_entity = self.metadata.get_by_name(entity=Table, fqn=from_fqn)
                 if from_entity:
-                    for to_fqn in self.get_outlets(task) or []:
+                    for to_fqn in xlet.outlets:
```

Comment:
Can we handle some fallbacks here, chances are it may come as None

<img width="537" alt="Screenshot 2023-03-15 at 5 31 25 PM" src="https://user-images.githubusercontent.com/40225091/225303166-40f750ef-d75d-49cc-8a7a-9f1859f27587.png">